source/jormungandr/jormungandr/scenarios/new_default.py

```diff
@@ -491,6 +495,52 @@ def update_total_co2_emission(pb_resp):
         j.co2_emission.unit = 'gEC'
 
 
+def update_disruptions_on_pois(instance, pb_resp):
+    """
+    Maintain a set of uri from all journey.section.origin and journey.section.destination of type Poi,
+    call loki with api api_disruptions&pois[]...
+    For each disruption on poi, add disruption id in the attribute links and add disruptions in the response
+    """
+    if not pb_resp.journeys:
+        return
+    # Add uri of all the pois in a set
+    poi_uris = set()
+    since_datetime = date_to_timestamp(datetime.utcnow())
+    until_datetime = date_to_timestamp(datetime.utcnow())
+    for j in pb_resp.journeys:
+        for s in j.sections:
+            if s.origin.embedded_type == type_pb2.POI:
+                poi_uris.add(s.origin.uri)
+                since_datetime = min(since_datetime, s.begin_date_time)
+
+            if s.destination.embedded_type == type_pb2.POI:
+                poi_uris.add(s.destination.uri)
+                until_datetime = max(until_datetime, s.end_date_time)
+
+    # Get disruptions for poi_uris calling loki with api poi_disruptions and poi_uris in param
+    poi_disruptions = get_disruptions_on_poi(instance, poi_uris, since_datetime, until_datetime)
+    if poi_disruptions is None:
+        return
+
+    # For each poi in the response:
+    # add impact_uris from resp_poi and
+    # copy object poi in impact.impacted_objects
+    for j in pb_resp.journeys:
+        for s in j.sections:
+            if s.origin.embedded_type == type_pb2.POI:
```

Comment:
you can keep the track of disrupted pt_object in a map, so you don't have re-iterate the whole response again